lib/pacemaker/pcmk_sched_colocation.c

```diff
@@ -1220,6 +1220,51 @@ add_node_scores_matching_attr(GHashTable *nodes, const pe_resource_t *rsc,
     }
 }
 
+/*!
+ * \internal
+ * \brief Initialize colocated node table for a group resource
+ *
+ * \param[in] rsc     Group resource being colocated with another resource
+ * \param[in] log_id  Resource ID to use in log messages
+ * \param[in] nodes   Nodes to update
+ * \param[in] attr    Colocation attribute (NULL to use default)
+ * \param[in] factor  Incorporate scores multiplied by this factor
+ * \param[in] flags   Bitmask of enum pcmk__coloc_select values
+ *
+ * \return Table of node scores initialized for colocation
+ */
+static GHashTable *
+init_group_colocated_nodes(const pe_resource_t *rsc, const char *log_id,
+                           GHashTable **nodes, const char *attr, float factor,
+                           uint32_t flags)
+{
+    GHashTable *work = NULL;
+    pe_resource_t *member = NULL;
+
+    if (*nodes == NULL) {
+        // Only cmp_resources() passes a NULL nodes table
+        member = pe__last_group_member(rsc);
```

Comment:
This is probably correct and it matches the existing logic, but the reasoning for choosing the last resource instead of the first in this case isn't immediately obvious to me. I haven't had time to do a deeper dive yet.

Your commit message in an earlier commit says that `pcmk__group_assign()` adds the group's colocations to the last member. But `cmp_resources()` appears to run before the assign function, so assign hasn't been called yet.

Reply:
I hadn't gotten as far as dealing with that yet. I suspect it has issues. The PR was going to be scary long if I kept going :)